core/ir/instr_api.h

```diff
@@ -2577,9 +2587,13 @@ instr_is_reg_spill_or_restore(void *drcontext, instr_t *instr, bool *tls DR_PARA
 #    define EFLAGS_WRITE_ARITH EFLAGS_WRITE_6
 
 /** Converts an EFLAGS_WRITE_* value to the corresponding EFLAGS_READ_* value. */
-#    define EFLAGS_WRITE_TO_READ(x) ((x) >> 11)
+#    define EFLAGS_WRITE_TO_READ(x)                              \
+        ((((x) & (EFLAGS_WRITE_ALL & ~EFLAGS_WRITE_AC)) >> 11) | \
+         (((x)&EFLAGS_WRITE_AC) >> 1))
```

Comment:
Safer to wrap all the EFLAGS_ constants in parens just in case they turn out to be an expression.

Ditto below.

Reply:
Done.